File: Chapter 1/Linear_search/Python/Linear search.py
def linear_search(list, item):

    for i in range(len(list)):
        if list[i] == item:
            return i


myarr = []
length = int(input("Ievadiet masiva garumu: "))
for i in range(length):
    myarr.append(int(input("Ievadiet elementu: ")))

item = int(input("Ievadiet meklejamo elementu: "))

print (linear_search(myarr, item))

File: Chapter 1/Linear_search/C++/Linear search.cpp
#include <iostream>

using namespace std;

int linear_search(int* arr, int lenght, int item) {

	for (int i = 0; i < lenght; i++)
	{
		if (arr[i] == item)
		{
			return i;
		}
		else return NULL;
	}
}

int main() {

	int lenght;
	int item;

	cout << "Ievadiet masiva garumu: ";
	cin >> lenght;
	cout << endl;

	int *elements = new int[lenght];

	cout << "Ievadiet " << lenght << " skaitlus: ";
		for (int i = 0; i < lenght; i++)
		{
			cout << " " << i + 1 << ". ";
			cin >> elements[i];
		}
	cout << endl;

	cout << "Ievadiet meklejamo elementu masiva: ";
	cin >> item;
	cout << endl;

	if (linear_search(elements, lenght, item) == NULL)
	{
		cout << "Tada elementa nav.";
	}
	else
	{
		cout << "Elementa numurs masiva: " << linear_search(elements, lenght, item);
	}

	return 0;
}

